usaspending_api/common/helpers/decorators.py

```diff
@@ -0,0 +1,72 @@
+import logging
+
+from django.conf import settings
+from django.db import connection
+from django.db.utils import OperationalError
+
+from usaspending_api.common.exceptions import EndpointTimeoutException
+
+logger = logging.getLogger(__name__)
+
+
+def set_db_timeout(timeout_in_seconds=settings.DEFAULT_DB_TIMEOUT_IN_SECONDS):
+    """ Decorator used to set the database statement timeout within the Django app scope
+
+        Args:
+            timeout_in_seconds: timeout value to set, in milliseconds
+
+        NOTE:
+            The statement_timeout is only set for this specific connection. Any subsequent request will have a default
+            value of 0. The alteration DOES NOT persist across connections and since each request is a new connection to
+            the database, there is no need to reset the statement_timeout value.
+
+        Examples:
+            @set_db_timeout(test_timeout_in_ms)
+            def func_running_db_call(...):
+                ...
+
+            OR
+
+            @set_db_timeout()  # This will use the default value in settings
+            def func_running_db_call(...):
+                ...
+    """
+
+    timeout_in_ms = int(timeout_in_seconds * 1000)
+
+    def wrap(func):
+        def wrapper(*args, **kwargs):
+            with connection.cursor() as cursor:
+                cursor.execute("show statement_timeout")
```

Comment:
Are you sure that our app db user will have permissions to query the timeout in prod?

Reply:
This is a good thought, thinking back, i'm not sure if we're even able to set the statement timeout on a read replica. the django app manages switching between the replica and main, so that'll need to be investigated further